catalogue_pipeline/matcher/src/main/scala/uk/ac/wellcome/platform/matcher/locking/DynamoRowLockDao.scala

```diff
@@ -37,46 +40,59 @@ class DynamoRowLockDao @Inject()(
   def lockRow(id: Identifier, contextId: String) = Future {
     val (created, expires) = getExpiry
     val rowLock = RowLock(id.id, contextId, created, expires)
-
-    debug(s"Trying to create RowLock: $rowLock")
+    debug(s"Locking $rowLock")
 
     val scanamoOps = table
       .given(
         not(attributeExists('id)) or Condition(
           'expires < rowLock.created.getEpochSecond))
       .put(rowLock)
-
     val result = Scanamo.exec(dynamoDBClient)(scanamoOps)
-
-    debug(s"Got $result when creating $rowLock")
+    debug(s"Got $result for $rowLock")
 
     result match {
       case Right(_) => rowLock
-      case Left(error) =>
-        throw new FailedLockException(s"Failed to lock $id", error)
+      case Left(error) => {
+        debug(s"Failed to lock $rowLock $error")
+        throw FailedLockException(s"Failed to lock $id", error)
+      }
```

Comment:
What happens if the call to `Scanamo.exec` throws an exception? I think, `lockRow` will return a failed future, but the wrapped exception will not be a `FailedLockException`.If we want metrics around failed locks and unlocks, we probably want it to be a `FailedLockException`. It could be useful to add a test around this scenario, perhaps? I think practically this is what happens with ThroughputEceededExceptions